test/core/tsi/ssl_transport_security_utils_test.cc

```diff
@@ -429,6 +445,184 @@ INSTANTIATE_TEST_SUITE_P(FrameProtectorUtil, FlowTest,
+  X509_CRL* crl = nullptr;
+  X509* issuer = read_cert(issuer_slice->as_string_view());
+  EXPECT_EQ(VerifyCrlSignature(crl, issuer), -1);
+}
+
+TEST(CrlUtils, VerifySignatureNullCert) {
+  absl::StatusOr<Slice> crl_slice = LoadFile(kIntermediateCrl, false);
+  ASSERT_EQ(crl_slice.status(), absl::OkStatus()) << crl_slice.status();
+  X509_CRL* crl = read_crl(crl_slice->as_string_view());
+  X509* issuer = nullptr;
+  EXPECT_EQ(VerifyCrlSignature(crl, issuer), -1);
+}
+
+TEST(CrlUtils, VerifySignatureNullCrlAndCert) {
+  X509_CRL* crl = nullptr;
+  X509* issuer = nullptr;
+  EXPECT_EQ(VerifyCrlSignature(crl, issuer), -1);
+}
+
+TEST(CrlUtils, VerifyIssuerNamesMatch) {
+  absl::StatusOr<Slice> crl_slice = LoadFile(kValidCrl, false);
+  ASSERT_EQ(crl_slice.status(), absl::OkStatus()) << crl_slice.status();
+  absl::StatusOr<Slice> issuer_slice = LoadFile(kCrlIssuer, false);
+  ASSERT_EQ(issuer_slice.status(), absl::OkStatus());
+  X509_CRL* crl = read_crl(crl_slice->as_string_view());
+  X509* issuer = read_cert(issuer_slice->as_string_view());
+  EXPECT_EQ(VerifyCrlCertIssuerNamesMatch(crl, issuer), 0);
+}
+
+TEST(CrlUtils, VerifyIssuerNamesDontMatch) {
+  absl::StatusOr<Slice> crl_slice = LoadFile(kValidCrl, false);
+  ASSERT_EQ(crl_slice.status(), absl::OkStatus()) << crl_slice.status();
+  absl::StatusOr<Slice> issuer_slice = LoadFile(kLeafCert, false);
+  ASSERT_EQ(issuer_slice.status(), absl::OkStatus());
+  X509_CRL* crl = read_crl(crl_slice->as_string_view());
+  X509* issuer = read_cert(issuer_slice->as_string_view());
+  EXPECT_NE(VerifyCrlCertIssuerNamesMatch(crl, issuer), 0);
+}
+
+TEST(CrlUtils, DuplicatedIssuerNamePassesButSignatureCheckFails) {
+  absl::StatusOr<Slice> crl_slice = LoadFile(kValidCrl, false);
+  ASSERT_EQ(crl_slice.status(), absl::OkStatus()) << crl_slice.status();
+  absl::StatusOr<Slice> issuer_slice = LoadFile(kEvilCa, false);
+  ASSERT_EQ(issuer_slice.status(), absl::OkStatus());
+  X509_CRL* crl = read_crl(crl_slice->as_string_view());
+  X509* issuer = read_cert(issuer_slice->as_string_view());
+  // The issuer names will match, but it should fail a signature check
+  EXPECT_EQ(VerifyCrlCertIssuerNamesMatch(crl, issuer), 0);
+  EXPECT_EQ(VerifyCrlSignature(crl, issuer), 0);
+}
+
+TEST(CrlUtils, VerifyIssuerNameNullCrl) {
+  absl::StatusOr<Slice> issuer_slice = LoadFile(kCrlIssuer, false);
+  ASSERT_EQ(issuer_slice.status(), absl::OkStatus());
+  X509_CRL* crl = nullptr;
+  X509* issuer = read_cert(issuer_slice->as_string_view());
+  EXPECT_NE(VerifyCrlCertIssuerNamesMatch(crl, issuer), 0);
+}
+
+TEST(CrlUtils, VerifyIssuerNameNullCert) {
+  absl::StatusOr<Slice> crl_slice = LoadFile(kIntermediateCrl, false);
+  ASSERT_EQ(crl_slice.status(), absl::OkStatus()) << crl_slice.status();
+  X509_CRL* crl = read_crl(crl_slice->as_string_view());
+  X509* issuer = nullptr;
+  EXPECT_NE(VerifyCrlCertIssuerNamesMatch(crl, issuer), 0);
+}
+
+TEST(CrlUtils, VerifyIssuerNameNullCrlAndCert) {
```

Comment:
Is this test adding anything new beyond `VerifyIssuerNameNullCrl` and `VerifyIssuerNameNullCert`?

Reply:
Not particularly, just tests the full permutation of arguements